### PR TITLE
Updates to HNN

### PR DESCRIPTION
**Pull request by avik-pal**
HNN is Lux Compatible + Use Forward-over-Reverse rather than Reverse-over-Reverse

### CODE REVIEW

**Comment by avik-pal** [comment]
Won't work till https://github.com/SciML/SciMLSensitivity.jl/pull/827 is merged.

File: docs/src/examples/hamiltonian_nn.md [hunk 1]
```diff
@@ -9,50 +9,46 @@ m\ddot x + kx = 0
...
+dataloader = ((selectdim(data, 2, ((i-1)*B+1):(min(i * B, size(data, 2)))))
+              for i in 1:(size(data, 2)÷B))
 
-loss(x, y, p) = mean((hnn(x, p) .- y) .^ 2)
+hnn = HamiltonianNN(Lux.Chain(Lux.Dense(2, 64, relu), Lux.Dense(64, 1)))
+ps, st = Lux.setup(Random.default_rng(), hnn)
+ps_c = ps |> ComponentArray
 
-callback() = println("Loss Neural Hamiltonian DE = $(loss(data, target, p))")
+opt = ADAM(0.01f0)
```
Comment (Vaibhavdixit02) [inline docs/src/examples/hamiltonian_nn.md:31]: Can you use OptimizationOptimisers instead?

Comment (avik-pal) [reply docs/src/examples/hamiltonian_nn.md:31]: I had run into some errors earlier with that. Let me try once again tomorrow and see.

Comment (avik-pal) [reply docs/src/examples/hamiltonian_nn.md:34]: The latest commit uses Optimization.jl

**Pull request merged**